dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -56,137 +74,181 @@
-        request = ListObjectsV2Request.builder().bucket(BUCKET_NAME).build();
-        listObjectsV2Response = s3Client.listObjectsV2(request);
-        contents = listObjectsV2Response.contents();
-        assertEquals(3, contents.size());
-        s3BucketKeys = contents.stream().map(S3Object::key).toList();
-        assertTrue(s3BucketKeys.contains("workflow/github.com/ENCODE-DCC/pipeline-container%2Fencode-mapping-cwl/1.0/terra/" + fileName));
-        assertTrue(s3BucketKeys.contains("workflow/github.com/ENCODE-DCC/pipeline-container%2Fencode-mapping-cwl/1.0/agc/" + fileName));
-        assertTrue(s3BucketKeys.contains("tool/quay.io/briandoconnor/dockstore-tool-md5sum/1.0/terra/" + fileName));
+        executions = createExecutions(2);
+        extendedGa4GhApi.executionMetricsPost(executions, platform2, workflowId, workflowVersionId, description);
+        metricsDataList = verifyMetricsDataList(workflowId, workflowVersionId, 2);
+        metricsData = verifyMetricsDataInfo(metricsDataList, workflowId, workflowVersionId, platform2, String.format(workflowExpectedS3KeyPrefixFormat, platform2));
+        verifyMetricsDataMetadata(metricsData, ownerUserId, description);
+        verifyMetricsDataContent(metricsData, executions);
+
+        // Register and publish a tool
+        DockstoreTool tool = new DockstoreTool();
+        tool.setDefaultCwlPath("/cwls/cgpmap-bamOut.cwl");
+        tool.setGitUrl("git@github.com:DockstoreTestUser2/dockstore-cgpmap.git");
+        tool.setNamespace("dockstoretestuser2");
+        tool.setName("dockstore-cgpmap");
+        tool.setRegistryString(Registry.QUAY_IO.getDockerPath());
+        tool.setDefaultVersion("symbolic.v1");
+        tool.setDefaultCWLTestParameterFile("/examples/cgpmap/bamOut/bam_input.json");
+        DockstoreTool registeredTool = containersApi.registerManual(tool);
+        registeredTool = containersApi.refresh(registeredTool.getId());
+        containersApi.publish(registeredTool.getId(), CommonTestUtilities.createOpenAPIPublishRequest(true));
+
+        final String toolId = "quay.io/dockstoretestuser2/dockstore-cgpmap";
+        final String toolVersionId = "symbolic.v1";
+        final String toolExpectedS3KeyPrefixFormat = "tool/quay.io/dockstoretestuser2/dockstore-cgpmap/symbolic.v1/%s"; // This is the prefix without the file name. Format by providing the platform
+        extendedGa4GhApi.executionMetricsPost(executions, platform1, toolId, toolVersionId, null);
+        metricsDataList = verifyMetricsDataList(toolId, toolVersionId, 1);
+        metricsData = verifyMetricsDataInfo(metricsDataList, toolId, toolVersionId, platform1, String.format(toolExpectedS3KeyPrefixFormat, platform1));
+        verifyMetricsDataMetadata(metricsData, ownerUserId, "");
+        verifyMetricsDataContent(metricsData, executions);
+        assertEquals(3, getS3ObjectsFromBucket().size(), "There should be 3 objects, 2 for workflows and 1 for tools");
```

Comment:
thanks for tackling both tools and workflows, while we're at it, do we think there will be any issues with workflow-like stuff like notebooks and services or should they just work?

Reply:
It's an extended GA4GH endpoint, so I think as long as notebooks and services have a TRS ID, then it should work. We would probably want to make the S3 key prefix different though. For example, notebooks go in a top-level `notebook` folder